### PR TITLE
Docs: Images in Notifications

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What this PR does / why we need it**:

This pull request adds documentation for images in notifications.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/alerting/images-in-notifications.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+title: Images in Notifications
+---
+
+# Images in Notifications
+
+Images in notifications helps recipients of alert notifications better understand why an alert has fired or resolved by including an image of the panel for the Grafana managed alert rule.
+
+> **Note**: Images in notifications are not available for Grafana Mimir and Loki managed alert rules, or when Grafana is set up to send alert notifications to an external Alertmanager.
+
+If Grafana is set up to send images in notifications it takes a screenshot of the panel for the Grafana managed alert rule when either of the following happen:
```
Comment (armandgrillet) [inline docs/sources/alerting/images-in-notifications.md:18]:
```suggestion
If Grafana is set up to send images in notifications, it takes a screenshot of the panel for the Grafana managed alert rule when either of the following happen:
```

File: docs/sources/alerting/images-in-notifications.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+> **Note**: It is recommended that an external image store is used for images in notifications as not all contact points supported uploading images from disk, and should `temp_data_lifetime` be less than the `group_wait` and `group_interval` options in Alertmanager then it is possible that the image on disk is deleted before an alert notification is even sent.
+
+## Requirements
+
+To use images in notifications Grafana must be set up to use [image rendering]({{< relref "../setup-grafana/image-rendering/" >}}). It is also recommended that Grafana is set up to upload images to an [external image store]({{< relref "../setup-grafana/configure-grafana/#external_image_storage" >}}) such as Amazon S3, Azure Blob Storage, Google Cloud Storage or even Grafana.
```
Comment (armandgrillet) [inline docs/sources/alerting/images-in-notifications.md:29]:
```suggestion
To use images in notifications, Grafana must be set up to use [image rendering]({{< relref "../setup-grafana/image-rendering/" >}}). It is also recommended that Grafana is set up to upload images to an [external image store]({{< relref "../setup-grafana/configure-grafana/#external_image_storage" >}}) such as Amazon S3, Azure Blob Storage, Google Cloud Storage or even Grafana.
```

File: docs/sources/alerting/images-in-notifications.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    # Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please
+    # see [external_image_storage] for further configuration options. If this option is false then
+    # screenshots will be persisted to disk for up to temp_data_lifetime.
+    upload_external_image_storage = false
+
+Restart Grafana for the changes to take affect.
+
+## Supported notifiers
+
+Images in notifications are supported in the following notifiers:
```
Comment (armandgrillet) [inline docs/sources/alerting/images-in-notifications.md:58]:
```suggestion
Images in notifications are supported in the following notifiers and additional support will be added in the future:
```

File: docs/sources/alerting/images-in-notifications.md (hunk 1 above)
Comment (brendamuir) [inline docs/sources/alerting/images-in-notifications.md:18]:
```suggestion
# Images in notifications

Images in notifications helps recipients of alert notifications better understand why an alert has fired or resolved by including an image of the panel for the Grafana managed alert rule.

> **Note**: Images in notifications are not available for Grafana Mimir and Loki managed alert rules, or when Grafana is set up to send alert notifications to an external Alertmanager.

If Grafana is set up to send images in notifications, it takes a screenshot of the panel for the Grafana managed alert rule when either of the following happen:
```

File: docs/sources/alerting/images-in-notifications.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+If Grafana is set up to send images in notifications it takes a screenshot of the panel for the Grafana managed alert rule when either of the following happen:
+
+1. The alert rule transitions from pending to firing
+2. The alert rule transitions from firing to OK
+
+Images are stored in the [data]({{< relref "../setup-grafana/configure-grafana/#paths" >}}) path and so Grafana must have write-access to this path. If Grafana cannot write to data then taken screenshots cannot be saved to disk and an error will be logged for each failed screenshot attempt. In addition to storing images on disk Grafana can also store the image in an external image store such as Amazon S3, Azure Blob Storage, Google Cloud Storage and even Grafana where screenshots are stored in `public/img/attachments`. Screenshots older than `temp_data_lifetime` are deleted from disk but not the external image store. If Grafana is the external image store then screenshots are deleted from `data` but not from `public/img/attachments`.
```
[the remark this answers is not in the document]
Comment (brendamuir) [reply docs/sources/alerting/images-in-notifications.md:23]: Agree with the above and also split first two sentences as they are long? Images are stored in the [data]({{< relref "../setup-grafana/configure-grafana/#paths" >}}) path.  Grafana must have write-access to this path.

File: docs/sources/alerting/images-in-notifications.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+To use images in notifications Grafana must be set up to use [image rendering]({{< relref "../setup-grafana/image-rendering/" >}}). It is also recommended that Grafana is set up to upload images to an [external image store]({{< relref "../setup-grafana/configure-grafana/#external_image_storage" >}}) such as Amazon S3, Azure Blob Storage, Google Cloud Storage or even Grafana.
+
+## Configuration
+
+If Grafana has been set up to use [image rendering]({{< relref "../setup-grafana/image-rendering/" >}}) images in notifications can be turned on via the `capture` option in `[unified_alerting.screenshots]`:
+
+    # Enable screenshots in notifications. This option requires a remote HTTP image rendering service. Please
+    # see [rendering] for further configuration options.
+    capture = true
+
```
Comment (brendamuir) [inline docs/sources/alerting/images-in-notifications.md:38]:
```suggestion
If Grafana has been set up to use [image rendering]({{< relref "../setup-grafana/image-rendering/" >}}), images in notifications can be turned on using the `capture` option in `[unified_alerting.screenshots]`:

    # Enable screenshots in notifications. This option requires a remote HTTP image rendering service. Please
    # see [rendering] for further configuration options.
    capture = true

```

File: docs/sources/alerting/images-in-notifications.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    # that can be taken at the same time for all firing alerts where as concurrent_render_request_limit sets
+    # the total number of concurrent screenshots across all Grafana services.
+    max_concurrent_screenshots = 5
+
+If Grafana has been set up to use an external image store then `upload_external_image_storage` should be set to `true`:
+
+    # Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please
+    # see [external_image_storage] for further configuration options. If this option is false then
+    # screenshots will be persisted to disk for up to temp_data_lifetime.
+    upload_external_image_storage = false
```
Comment (brendamuir) [inline docs/sources/alerting/images-in-notifications.md:52]:
```suggestion
If Grafana has been set up to use an external image store, `upload_external_image_storage` should be set to `true`:

    # Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please
    # see [external_image_storage] for further configuration options. If this option is false then
    # screenshots will be persisted to disk for up to temp_data_lifetime.
    upload_external_image_storage = false
```

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 7]
```diff
@@ -1259,6 +1259,24 @@ The interval string is a possibly signed sequence of decimal numbers, followed b
...
+
+Enable screenshots in notifications. This option requires a remote HTTP image rendering service. Please see `[rendering]` for further configuration options.
+
+### max_concurrent_screenshots
+
+The maximum number of screenshots that can be taken at the same time. This option is different from `concurrent_render_request_limit` as `max_concurrent_screenshots` sets the number of concurrent screenshots that can be taken at the same time for all firing alerts where as concurrent_render_request_limit sets the total number of concurrent screenshots across all Grafana services.
+
+### upload_external_image_storage
+
+Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please see `[external_image_storage]` for further configuration options. If this option is false then screenshots will be persisted to disk for up to `temp_data_lifetime`.
```
Comment (brendamuir) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1276]:
```suggestion
Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please see `[external_image_storage]` for further configuration options. If this option is false, screenshots will be persisted to disk for up to `temp_data_lifetime`.
```

**Review by brendamuir — APPROVED** [review]

**Pull request merged**